src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +325,124 @@ public void addSeparator() {
+							}
+						}
+					});
+					
+					reEstablishConnection.execute();
+				});
+			}
+			return btnReopen;
+		}
+		
+		private JButton getBtnReopenEdit(WebSocketPanelSender sender){
+			if(btnReopenEdit == null) {
+				btnReopenEdit = new JButton();
+				btnReopenEdit.setIcon(REOPEN_EDIT_ICON);
+				btnReopenEdit.setEnabled(true);
+				btnReopenEdit.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_edit_hint"));
+				btnReopenEdit.addActionListener(e -> {
+					try {
+						wsUiHelper.setSecWebSocketKeyField(sender.getWebSocketKey(wsUiHelper.getSelectedChannelId()));
+						JOptionPane.showMessageDialog(this, getReopenPanel(),
+								Constant.messages.getString("websocket.manual_send.adv_dialog.title"),
+								JOptionPane.PLAIN_MESSAGE);
+					} catch (IllegalArgumentException e1) {
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.invalid_channel"));
+					} catch (Exception e1) {
+						LOGGER.error(e1,e1);
+						View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.unable_reopen"));
+					}
+				});
+			}
+			return btnReopenEdit;
+		}
+
+		private JPanel getReopenPanel(){
+			if(reopenPanel == null){
+				reopenPanel = new JPanel();
+				reopenPanel.setLayout(new GridBagLayout());
+				int y = 0;
+
+				reopenPanel.add(wsUiHelper.getRedirectionCheckBox(),wsUiHelper.getFieldConstraints(0,y++));
+				reopenPanel.add(wsUiHelper.getTrackingSessionCheckBox(),wsUiHelper.getFieldConstraints(0,y++));
+
+				reopenPanel.add(wsUiHelper.createVerticalSeparator(),wsUiHelper.getFieldConstraints(0,y++));
+				reopenPanel.add(wsUiHelper.getWebSocketKeyLabel(),wsUiHelper.getLabelConstraints(0,y++));
+				reopenPanel.add(wsUiHelper.getWebSocketKeyField(),wsUiHelper.getFieldConstraints(0,y++));
+				reopenPanel.add(wsUiHelper.getAlwaysGenerateCheckBox(),wsUiHelper.getFieldConstraints(0,y++));
+				reopenPanel.add(wsUiHelper.getGenerateWebSocketKeyButton(),wsUiHelper.getFieldConstraints(0,y++));
+
+			}
+			return reopenPanel;
+		}
+		
+		private void isOnConnectingProgress(boolean isOnProgress){
+			btnReopen.setEnabled(!isOnProgress);
+			btnReopenEdit.setEnabled(!isOnProgress);
+			connectingIcon.setVisible(isOnProgress);
+			connectingLabel.setVisible(isOnProgress);
+		}
+		
+		class ReEstablishConnection extends SwingWorker<Integer,Boolean> {
+			private String newSecWebSocketKey;
+
+        	protected ReEstablishConnection(String newSecWebSocketKey){
+				this.newSecWebSocketKey = newSecWebSocketKey;
+			}
+			
+        	public static final int IOEXCEPTION = -1;
+			public static final int REQUEST_OUT_OF_SCOPE = -2;
+			
+			@Override
+			protected Integer doInBackground(){
+				int channelId;
+				try {
+					channelId = sender.reOpenChannel(wsUiHelper.getSelectedChannelId(), newSecWebSocketKey, wsUiHelper.isRedirection(), wsUiHelper.isTrackingSession());
+				} catch (IOException e) {
+					LOGGER.info(e.getStackTrace(),e);
```

Comment:
- This should probably be debug (this is expected to happen).
 - Same comment as above regarding the message.